code/datums/achievements/_achievement_data.dm

```diff
@@ -71,6 +71,15 @@
 	else if(istype(A, /datum/award/score))
 		data[achievement_type] += 1
 
+/datum/achievement_data/proc/increase_score(datum/award/score/achievement_type, mob/user, value)
+	var/datum/award/score/A = SSachievements.awards[achievement_type]
+	get_data(achievement_type) //Get the current status first if necessary
+	if(A.high_scores[1] < value)
+		for(var/client/C in GLOB.clients)
+			to_chat(C, "<span class='greenannounce'><B>[user.client.key] set a new high score in [A.name]: [value]</B></span>")
+	if(value > data[achievement_type])
+		data[achievement_type] = value
```

Comment:
Is it possible that data doesn't contain achievement_type and could have an index out of bounds exception?

Reply:
No, data contains all scores, even if you don't have them.